### PR TITLE
Part 1/2 of Final Changes

### PR DESCRIPTION
**Pull request by BashirKad**
These are all the changes I made to the Library class. I tried to keep it to a minimum, but there were a few things I had to change to get GUI to work and a few things that changing just made it easier to code.

The changes I made include: (TL;DR at the bottom)
--> Completely rewriting of the checkOut() method (it is now found in the GUI class)
--> I created a new createPerson() method that is slightly different, I kept the old one and named the new one createPerson2(), but once everyone has looked at this I'll 
      make one final commit after this deleting the commented code
--> most of the main classes functionality was moved to GUI class
--> On the topic of functionality, everyone should work the same as before, but if I'm missing any features please tell me, but I made sure to keep it the same as before
--> The rest of the changes are in the GUI class and I'll upload that in a minute

TL;DR for this entire thing: 
huge blocks of commented code were rewritten/moved, most of the code is in GUI now, functionality should still be the same

### CODE REVIEW

**Comment by MariaMa-GitHub** [comment]
@BashirKad I have a question: did you update your version of the code? cuz we made some changes and you kinda have to take that into account

**Comment by MariaMa-GitHub** [comment]
@BashirKad if possible, you can **fork this repository** (so you can make all the changes that you want) and work on that copy. Make sure that your GUI code works well with what we currently have and then you can request to merge your forked version to the current one (if not possible to merge, you can just upload files)

**Comment by BashirKad** [comment]
I'm pretty sure I updated the majority of the code, but I may have missed something and didn't realize

**Comment by BashirKad** [comment]
I assume this is what you mean, but in my IDE all my other classes are the exact same as the classes in the repo, the only different one is Library and the added GUI class obviously

**Comment by MariaMa-GitHub** [comment]
Kk then you can run both the current version and your version on your end
to compare them (see whether or not you have missed something). You can
also look at the history of the repository’s commits (there you can see all
the changes we’ve made)

**Comment by MariaMa-GitHub** [comment]
The subclasses and Media class too ? I changed the constructors (I think
those are my most recent changes)

**Comment by BashirKad** [comment]
yeah I updated them all

**Comment by BashirKad** [comment]
I just looked through the code again and I'm 99% sure I have everything correctly in place. @bobyangg @actualbobbernauts if either of you are still online, do you mind just going through the merge requests to make sure I'm not forgetting anything (and just to make sure we're all on the same page).
